src/bin/fireguard.rs

```diff
@@ -6,7 +6,10 @@ use color_eyre::eyre::Result;
 
 use fireguard::run;
 
-#[tokio::main]
+/// Since reqwest uses a tokio task underneath, I believe we need at
+/// least 2 available threads to ensure we can run both the upgrade task
+/// and the loop waiting for signals asyncronously
+#[tokio::main(worker_threads = 2)]
```

Comment:
Yep, this looks to be the issue we ran into for most of the afternoon!

Reply:
interesting! i wonder if there is a way of either detecting this based on the number of cpu cores or, even better, if there is a smart configuration for tokio to automatically do this